### PR TITLE
Provide capability to specify custom transaction id user property to get correlation info

### PR DESCRIPTION
**Pull request by tomkerkhove**
Signed-off-by: Tom Kerkhove <kerkhove.tom@gmail.com>

Provide capability to specify custom transaction id user property to get correlation info

### CODE REVIEW

File: src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs [hunk 1]
```diff
@@ -35,11 +35,12 @@ public static TProperty GetUserProperty<TProperty>(this Message message, string
         ///     Gets the correlation information for a given message.
         /// </summary>
         /// <param name="message">The received message.</param>
-        public static MessageCorrelationInfo GetCorrelationInfo(this Message message)
+        /// <param name="transactionIdPropertyName">Name of the property to determine the transaction id.</param>
+        public static MessageCorrelationInfo GetCorrelationInfo(this Message message, string transactionIdPropertyName = PropertyNames.TransactionId)
```
Comment (stijnmoreels) [inline src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs:39]: This should propably be guarded.

File: src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs [hunk 2]
```diff
@@ -60,11 +61,12 @@ private static string DetermineOperationId(string messageCorrelationId)
         /// <summary>	
         ///     Gets the transaction id that is linked to this message	
         /// </summary>	
-        /// <param name="message">Message to process</param>	
+        /// <param name="message">Message to process</param>
+        /// <param name="transactionIdPropertyName">Name of the property to determine the transaction id.</param>
         /// <returns>Transaction id for message</returns>
-        public static string GetTransactionId(this Message message)
+        public static string GetTransactionId(this Message message, string transactionIdPropertyName = PropertyNames.TransactionId)
```
Comment (stijnmoreels) [inline src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs:67]: This should propably be guarded.

File: src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs (hunk 1 above)
Comment (stijnmoreels) [inline src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs:39]: We'll have to be careful that this doesn't break dependent librarires. When a library only updates this core package and not the other packages, we may get a `MethodNotFound` exception along the way.

File: src/Arcus.Messaging.Tests.Unit/ServiceBus/MessageExtensionsTests.cs [hunk 3]
```diff
@@ -94,6 +94,27 @@ public void GetMessageCorrelationInfo_WithTransactionIdAsUserProperty_ReturnsCor
             Assert.Equal(expectedTransactionId, correlationInfo.TransactionId);
         }
 
+        [Fact]
+        public void GetMessageCorrelationInfo_WithTransactionIdInCustomUserProperty_ReturnsCorrelationInfoWithGeneratedOperationId()
```
Comment (stijnmoreels) [inline src/Arcus.Messaging.Tests.Unit/ServiceBus/MessageExtensionsTests.cs:98]: An test where we get the custom transaction ID directly can be appropriate also.

File: src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs (hunk 1 above)
Comment (stijnmoreels) [inline src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs:39]: I'm thinking, is there some place in the docs that this needs to be updated? I think there is.

**Review by stijnmoreels — CHANGES_REQUESTED** [review]

File: src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs (hunk 1 above)
Comment (tomkerkhove) [reply src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs:39]: Doesn't look like it 🤔

File: src/Arcus.Messaging.Tests.Unit/ServiceBus/MessageExtensionsTests.cs (hunk 3 above)
Comment (tomkerkhove) [reply src/Arcus.Messaging.Tests.Unit/ServiceBus/MessageExtensionsTests.cs:98]: What would that be then? Not sure if I get the suggestion.

File: src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs (hunk 1 above)
Comment (tomkerkhove) [reply src/Arcus.Messaging.ServiceBus.Core/Extensions/MessageExtensions.cs]: I've split them!

File: src/Arcus.Messaging.Tests.Unit/ServiceBus/MessageExtensionsTests.cs (hunk 3 above)
Comment (stijnmoreels) [reply src/Arcus.Messaging.Tests.Unit/ServiceBus/MessageExtensionsTests.cs:98]: Well, there was two methods updated but only one updated method is tested here, right? Or am I wrong?

Comment (tomkerkhove) [reply src/Arcus.Messaging.Tests.Unit/ServiceBus/MessageExtensionsTests.cs:98]: We don't have tests for the other one but will add thme

**Pull request merged**